fiftyone/core/odm/mixins.py

```diff
@@ -1118,6 +1125,21 @@ def _delete_field_schema(cls, path):
         doc._undeclare_field(field_name)
         _delete_field_doc(field_docs, field_name)
 
+    @classmethod
+    def _rename_indexes(cls, paths, new_paths):
+        updates = _get_index_updates(cls._dataset, paths, new_paths=new_paths)
+
+        for name, new_index_spec in updates.items():
+            cls._dataset.drop_index(name)
```

Comment:
A little bit sad that we have to do this instead of renaming the actual index datastructure, but this is another datapoint for @ritch ; in the future field changes (adding/renaming/deleting) can be treated as db migration that can be scheduled to run overnight or at a specified time to minimize disruption.